src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -469,26 +458,9 @@ long nextInterval(long now) {
                 return now + registrationIntervalMillis;
               }
 
-              boolean isWorkerPausedFromNewWork() {
```

Comment:
deleting of this mechanism might deserve some noting in the summary of the PR and future release notes

Reply:
I agree that this is hacky and likely not used by anyone. But the fact that the new mechanism only work on aws with lifecycle hook, i'm wondering if we want to keep this functionalities so that users on other cloud platforms can still use graceful shutdown through this.

Reply:
Ok i'll re-add this back in.